api/include/opentelemetry/trace/tracer-registry.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "./tracer.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+class Tracer;
+class TracerRegistry
+{
+ public:
+  static TracerRegistry* getInstance();
+  Tracer* const get(const std::string&);
```

Comment:
We've discussed to avoid having STL types in public interfaces to avoid object/vtable layout issue.